core/chain/abci.go

```diff
@@ -233,3 +274,35 @@ func (app *CoreApplication) isValidProtoEvent(tx []byte) (*gen_proto.Event, erro
 	}
 	return &msg, nil
 }
+
+func (app *CoreApplication) isValidSlaRollup(tx []byte) bool {
+	var msg gen_proto.SlaRollup
+	err := proto.Unmarshal(tx, &msg)
+	return err == nil
+}
+
+func (app *CoreApplication) validateBlockTxs(ctx context.Context, blockTime time.Time, blockHeight int64, txs [][]byte) (bool, error) {
+	alreadyContainsRollup := false
+	app.startInProgressTx(ctx)
+	for _, tx := range txs {
+		valid, err := app.isValidRollupTx(ctx, blockTime, blockHeight, tx)
+		if err != nil {
+			return false, err
+		} else if valid && alreadyContainsRollup {
+			return false, nil
+		} else if valid {
+			alreadyContainsRollup = true
+		} else if _, err := app.isValidProtoEvent(tx); err == nil {
+			continue
+		} else if app.isValidKVTx(tx) {
+			continue
+		} else {
+			return false, nil
+		}
+	}
+	if err := app.commitInProgressTx(ctx); err != nil {
+		app.logger.Error("Failure closing db while validating txs", "error", err)
+		return false, err
+	}
```

Comment:
We should not be committing anywhere in the abci outside of the commit call. What is being persisted here?